### PR TITLE
Add background information on multi site, multi instance and multi tenancy

### PR DESCRIPTION
**Pull request by allcaps**
Add background information on multi site, multi instance and multi tenancy

[Rendered](https://wagtail--7228.org.readthedocs.build/en/7228/advanced_topics/multi_site_multi_instance_multi_tenancy.html)

* Do the tests still pass? Y
* Does the code comply with the style guide? Y
* For Python changes: Have you added tests to cover the new/fixed behaviour? N/A
* For front-end changes: Did you test on all of [Wagtail’s supported environments] N/A
* For new features: Has the documentation been updated accordingly? N/A

### CODE REVIEW

File: docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+User, groups, and permissions can be configured in such a way that content creators can only manage the pages, images, and documents of a specific site. Wagtail can have multiple *site objects* and multiple *page trees*. Permissions can be linked to a specific page tree or a subsection thereof. Collections are used to categorize images and documents. A collection can be restricted to users who are in a specific group.
+
+Some projects require content editors to have permissions on specific sites and restrict access to other sites. Splitting *all* content per site and guaranteeing that no content 'leaks' is difficult to realize in a multi-site project. If you require full separation of content, then multi-instance might be a better fit...
+
+Multi-instance
+--------------
+
+Multi-instance is a Wagtail project configuration where a single code base is used by multiple websites. Each website has its server process, settings file, and a dedicated database and media directory. This guarantees the total separation of all content.
```
Comment (ababic) [inline docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: each website has its own

Comment (ababic) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: Also... Is there any particular reason why separate settings files are needed here? I would've thought you could get mostly the same level of control using a single one, with configuration at the env level (which would require no code changes to deploy a new site).

Comment (allcaps) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: The point is, that you can run with multiple settings. Maybe that is possible with environment variables. But adding env vars to the mix doesn't make it clearer what happens. 

Also, env vars are always the same right? How would you differentiate? Maybe 'a single code base' isn't clear enough. I mean, one directory on the server containing the source code, powering all instances. Not a single repo cloned in multiple directories.

File: docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  - arbitrary sections of the page tree
+  - sections of the collection tree (coming soon)
+  - one or more collections of documents and images
+
+- The page API is automatically scoped to the host used for the request
+
+But several features do not currently support multi-tenancy:
+
+- Snippets are global pieces of content so not suitable for multi-tenancy but any model that can be registered as a snippet can also be managed via the Wagtail model admin. You can add a site_id to the model and then use the model admin get_queryset method to determine which site can manage each object. The built-in snippet choosers can be replaced by `modelchooser <https://pypi.org/project/wagtail-modelchooser/>`_ that allows filtering the queryset to restrict which sites may display which objects.
+- Site, site setting, user, and group management. At the moment, your best bet is to only allow superusers to manage these objects.
```
Comment (ababic) [inline docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:71]: Maybe we could add something here as to why this is best left to superusers? Maybe "Permission configuration for built-in models like Sites, Site settings and Users are not site-specific, so any user with permission to edit a single entry can edit them all. This limitation can be mostly circumvented by only allowing superusers to manage these models.

File: docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst (hunk 1 above)
Comment (ababic) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: Okay, maybe I'm misunderstanding this setup in terms of how it actually runs. How do you control which settings file is used by each process?

Comment (ababic) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: I was imagining some exotic Kubernetes setup with the same docker image powering differently configured nodes.

Comment (allcaps) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: With uWSGI, the acom.ini contains 
`env = DJANGO_SETTINGS_MODULE=settings.acom`
And bcom.ini contains:
`env = DJANGO_SETTINGS_MODULE=settings.bcom`

With a development server, you would run 
`./manage.py 8000 --settings settings.acom`
`./manage.py 8001 --settings settings.bcom`

Using environment variable wound't get you very far. All is executed in the same directory on exactly the same code. Wouldn't the env vars would be similar?

Of course the cost of cloning the repo a second time and have your project code multiple times on disk isn't very expensive. Disk space is cheep. But deploying many instances might be extra work. A single project dir running multiple sites, means a single deploy.

Does that make sense?

Comment (ababic) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:29]: Yes, that makes sense. I think we could maybe just do with a tweak a here to make that clearer in the text?

File: docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst (hunk 2 above)
Comment (allcaps) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst:77]: Done.

File: docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst (hunk 1 above)
Comment (allcaps) [reply docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst]: @ababic I improved this section. Can you have another look?

**Review by ababic — APPROVED** [review]
LGTM 👍 thank you for your awesomeness!

**Comment by gasman** [comment]
Merged in 3f24beaae60e281417b7818c5204c447c7989885

**Pull request closed without merging**